lib/gems/pending/util/miq_ftp_lib.rb

```diff
@@ -0,0 +1,70 @@
+require 'net/ftp'
+require 'logger'
+
+# Helper methods for net/ftp based classes and files.
+#
+# Will setup a `@ftp` attr_accessor to be used as the return value for
+# `.connect`, the main method being provided in this class.  Will also setup
+# logging if not already done for the particular class (that follows the
+# VmdbLogger conventions, and setup a `uri` attr_accessor if that doesn't
+# already exist.
+module MiqFtpLib
+  def self.included(klass)
+    klass.send(:attr_accessor, :ftp)
+
+    klass.send(:attr_accessor, :uri) unless klass.instance_methods.include?(:uri=)
```

Comment:
At first glance it feels like there should be a better way to check if there is already an `attr_accessor` for this. If nothing else maybe check for both the `#uri=` and `#uri` methods?

Reply:
Yeah, I thought so too.  Again, this is a straight rip off of what I did in https://github.com/ManageIQ/manageiq/pull/17549 (simply changed and relocated the module), so I didn't change anything there.

Reply:
Yeah, can we simplify this.  Conditional attr_accessors feels complicated.  I don't understand why it's conditional.

Reply:
@jrafanie Sorry, missed this when looking at this earlier, but please look at https://github.com/ManageIQ/manageiq-gems-pending/pull/360#issuecomment-417097055 for clarity on to why this is the way.  Specifically the first bullet point.